rootex/framework/systems/transform_system.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include "core/renderer/renderer.h"
+#include "core/renderer/render_pass.h"
+#include "core/resource_files/basic_material_resource_file.h"
+#include "main/window.h"
+#include "framework/ecs_factory.h"
+#include "framework/scene.h"
+#include "framework/system.h"
+#include "framework/components/visual/camera_component.h"
+#include "components/visual/model/model_component.h"
+#include "components/visual/model/animated_model_component.h"
+#include "components/game/spring_arm_component.h"
```

Comment:
We just need `system.h`, and nothing else. Probably `scene.h` if it starts giving compile errors, but in general, try to reduce the number of includes as small as possible. The cost of every `#include` we do in a file is ginormous